Library/ViewModels/NoShippingPledgeViewModel.swift

```diff
@@ -987,32 +987,53 @@ public class NoShippingPledgeViewModel: NoShippingPledgeViewModelType, NoShippin
 
     // MARK: Pledge Over Time
 
-    self.showPledgeOverTimeUI = project.signal
+    let pledgeOverTimeUIEnabled = project.signal
       .map { ($0.isPledgeOverTimeAllowed ?? false) && featurePledgeOverTimeEnabled() }
 
-    self.pledgeOverTimeConfigData = Signal.combineLatest(
-      self.showPledgeOverTimeUI,
-      project,
-      pledgeTotal
+    let pledgeOverTimeQuery = Signal.combineLatest(project, pledgeTotal)
+      // Only make the query when PLOT is enabled
+      .filterWhenLatestFrom(pledgeOverTimeUIEnabled, satisfies: {
+        $0 == true
+      })
+      // Only call the query once
+      .take(first: 1)
+      .switchMap { (project: Project, pledgeTotal: Double) -> SignalProducer<
+        Signal<GraphAPI.BuildPaymentPlanQuery.Data, ErrorEnvelope>.Event,
+        Never
+      > in
+        let nf = NumberFormatter()
+        let amount = nf.string(from: NSNumber(value: pledgeTotal)) ?? ""
+        return AppEnvironment.current.apiService.buildPaymentPlan(
+          projectSlug: project.slug,
+          pledgeAmount: amount
+        ).materialize()
+      }
+
+    self.showPledgeOverTimeUI = Signal.merge(
```

Comment:
@jovaniks My assumption is that, since this feature is so small, there's no designs for a separate error state. Is that correct? My approach was just to hide the entire element if an error occurs in the `BuildPaymentPlan` query.

Reply:
That’s correct, we don’t have error handling planned for this component at the moment. Do you think it would make sense to add a `TODO` to revisit this discussion once everyone is back in January

Reply:
I'll set myself a reminder in Slack for January! And I'll raise it in the next PLOT meeting.